src/collectors/genericfile/genericfile.py

```diff
@@ -0,0 +1,84 @@
+# coding=utf-8
+"""
+This class collects data in key / value form from generic files.
+
+The collector operates on each line of a file, expecting the line to match the
+format provided by a configurable regular expression. It behaves similarly
+to awk.
+
+By default, the regular expression represents a common file pattern of key
+values pairs separated by either space, colon or equals (with optional
+surrounding space). The key value pairs can be any alphanumeric character.
+
+If customization is required, you can provide a different regex. Furthermore,
+you can override the match method if you need to do any further processing.
+
+#### Dependencies
+
+* re
+
+
+
+"""
+import diamond.collector
+import os
+import re
+
+
+class GenericFileCollector(diamond.collector.Collector):
+
+    def get_default_config_help(self):
+        config_help = super(GenericFileCollector,
+                            self).get_default_config_help()
+        config_help.update({
+            'regex': 'Defines the line format of the file.',
+            'file': 'The file to collect the data.',
+        })
+        return config_help
+
+    def get_default_config(self):
+        """
+        Returns the default collector settings
+        """
+        config = super(GenericFileCollector, self).get_default_config()
+        config.update({
+            'regex': '^\s*(?P<name>\w+)\s*[\s|:|=]\s*(?P<value>\w+)$',
+            'path': '',
```

Comment:
Yeah. I'm not sure what a good default path would be as the path will be dependent on the file. For example, if you're working with '/proc/meminfo', you could set the path to 'meminfo'.  I could give it some generic name, or remove it and treat it as a mandatory option that the user must set in the configuration.